service/history/execution/state_builder.go

```diff
@@ -132,7 +132,9 @@ func (b *stateBuilderImpl) ApplyEvents(
 		case types.EventTypeWorkflowExecutionStarted:
 			attributes := event.WorkflowExecutionStartedEventAttributes
 			var parentDomainID *string
-			if attributes.ParentWorkflowDomain != nil {
+			if attributes.ParentWorkflowDomainID != nil {
+				parentDomainID = attributes.ParentWorkflowDomainID
+			} else if attributes.ParentWorkflowDomain != nil {
```

Comment:
only domain or domain ID will be set in the attribute?

Reply:
Both would be set. But for older history `ParentWorkflowDomainID` is not available, thus it would fallback to previous logic.

Reply:
Added a comment.